### PR TITLE
Basic section

### LINKED ISSUE #6: Basics - Modes
- [x] Explain the modes;
- [x]  Explain the Normal Mode;
- [x]  Explain the Visual Mode;
- [x]  Explain the Insert Mode;

### PR DESCRIPTION
**Pull request by renatosuero**
It resolves #6 , @rgerardi  please when you can, review it . Thanks a lot =)

### CODE REVIEW

File: content/basics/vimmodes/_index.md [hunk 1]
```diff
@@ -4,4 +4,16 @@ date =  2017-10-30T10:09:17-04:00
 weight = 5
 +++
 
-Vim Modes
+The first thing that you need to know is, why Vim has modes... So when vim was created didn't exist GUI and mouse.
```
Comment (rgerardi) [inline content/basics/vimmodes/_index.md]: The first thing that you need to know about *vim* is: Vim is a modal editor. This means that *vim* has different modes that can be used to interact with it.

If you are asking why, remember that when the original **vi** editor was created in 1976, there were no graphical interfaces (GUI) or mouse, and even displays were not as powerful as todays.

File: content/basics/vimmodes/_index.md [hunk 2]
```diff
@@ -4,4 +4,16 @@ date =  2017-10-30T10:09:17-04:00
...
+Now let's think how I can write something or navigate in the lines... That's why you need to have modes and you need to know how to use it.
```
Comment (rgerardi) [inline content/basics/vimmodes/_index.md]: Now, in order to edit text or navigate around, you need to switch between the different modes.

File: content/basics/vimmodes/_index.md [hunk 3]
```diff
@@ -4,4 +4,16 @@ date =  2017-10-30T10:09:17-04:00
 weight = 5
 +++
 
-Vim Modes
+The first thing that you need to know is, why Vim has modes... So when vim was created didn't exist GUI and mouse.
+Now let's think how I can write something or navigate in the lines... That's why you need to have modes and you need to know how to use it.  
+Let's go to understand theses modes.
```
Comment (rgerardi) [inline content/basics/vimmodes/_index.md]: Let's understand the different modes available in **vim**:

File: content/basics/vimmodes/_index.md [hunk 4]
```diff
@@ -4,4 +4,16 @@ date =  2017-10-30T10:09:17-04:00
...
 +++
 
-Vim Modes
+The first thing that you need to know is, why Vim has modes... So when vim was created didn't exist GUI and mouse.
+Now let's think how I can write something or navigate in the lines... That's why you need to have modes and you need to know how to use it.  
+Let's go to understand theses modes.  
+
+------------
+
+- **Normal Mode**: *(as know as Command Mode)*, here is where you can do things like (copy,paste,find,replace) and execute commands like (:w to save,:q to quit).
```
Comment (rgerardi) [inline content/basics/vimmodes/_index.md]: **Normal Mode**: *(a.k.a Command Mode)*: here is where you can do things like copy, paste, find, or replace, and execute commands like (:w to save,or :q to quit).

File: content/basics/vimmodes/_index.md [hunk 5]
```diff
@@ -4,4 +4,16 @@ date =  2017-10-30T10:09:17-04:00
...
+
+------------
+
+- **Normal Mode**: *(as know as Command Mode)*, here is where you can do things like (copy,paste,find,replace) and execute commands like (:w to save,:q to quit). 
+- **Visual Mode**: here is where you can select text.   
+- **Insert Mode**: here is where you can edit your text.  
+
+------------
+
+> An important tip, look at the bottom vim and you'll see the mode where you are.You will see **INSERT** or **VISUAL** on left side, when you won't see it's because you are in **NORMAL/COMMAND** mode.
```
Comment (rgerardi) [inline content/basics/vimmodes/_index.md]: > An important tip: look at the bottom left of **vim's** screen and you'll see which mode is currently in use. You can see **INSERT** or **VISUAL** for the respective modes, or when you don't see anything it means you are in **NORMAL/COMMAND** mode.

File: content/basics/vimmodes/insert.md [hunk 6]
```diff
@@ -4,4 +4,18 @@ date =  2017-10-30T10:48:58-04:00
 weight = 6
 +++
 
-Insert Mode
+Insert Mode is where you can write/edit your documments. There are some keys to enter in this mode, let's see them:
```
Comment (rgerardi) [inline content/basics/vimmodes/insert.md]: Insert Mode is where you can write/edit your documments. From **Normal/Command** mode, you can use one of the following keys to switch to **Insert** mode:

File: content/basics/vimmodes/insert.md [hunk 7]
```diff
@@ -4,4 +4,18 @@ date =  2017-10-30T10:48:58-04:00
 weight = 6
 +++
 
-Insert Mode
+Insert Mode is where you can write/edit your documments. There are some keys to enter in this mode, let's see them:  
+
+| Key     | Description   																											|
+|---------|---------------------------------------------------------------------|
+| **i**   | This key enter in insert mode before under the cursor								|
```
Comment (rgerardi) [inline content/basics/vimmodes/insert.md]: | **i**   | This key switches to insert mode and allows you to edit text right before the current cursor position |

File: content/basics/vimmodes/insert.md [hunk 8]
```diff
@@ -4,4 +4,18 @@ date =  2017-10-30T10:48:58-04:00
...
+| **I**   | This key enter in insert mode begin line under the cursor						|
```
Comment (rgerardi) [inline content/basics/vimmodes/insert.md]: | **I**   | This key switches to insert mode and places the cursor at the beginning of the line						|

File: content/basics/vimmodes/insert.md [hunk 9]
```diff
@@ -4,4 +4,18 @@ date =  2017-10-30T10:48:58-04:00
...
+| **a**   | This key enter in insert mode after under the cursor 								|
```
Comment (rgerardi) [inline content/basics/vimmodes/insert.md]: | **a**   | This key switches to insert mode and allows you to edit text right after the current cursor position |

File: content/basics/vimmodes/insert.md [hunk 10]
```diff
@@ -4,4 +4,18 @@ date =  2017-10-30T10:48:58-04:00
...
+| **A**   | This key enter in insert mode end line under the cursor  						|
```
Comment (rgerardi) [inline content/basics/vimmodes/insert.md]: | **A**   | This key switches to insert mode and places the cursor at the end of the line  						|

File: content/basics/vimmodes/insert.md [hunk 11]
```diff
@@ -4,4 +4,18 @@ date =  2017-10-30T10:48:58-04:00
...
+| **o**   | This key enter in insert mode on the next line under the cursor  		|
```
Comment (rgerardi) [inline content/basics/vimmodes/insert.md]: | **o**   | This key switches to insert mode adding a line below the current one  		|

File: content/basics/vimmodes/insert.md [hunk 12]
```diff
@@ -4,4 +4,18 @@ date =  2017-10-30T10:48:58-04:00
...
+| **O**   | This key enter in insert mode on the previous line under the cursor |
```
Comment (rgerardi) [inline content/basics/vimmodes/insert.md]: | **O**   | This key switches to insert mode adding a line above the current one  		|

File: content/basics/vimmodes/insert.md [hunk 13]
```diff
@@ -4,4 +4,18 @@ date =  2017-10-30T10:48:58-04:00
...
+
+------------
+
+> Just to remember, if you need to move between words or lines, you need to move to **Normal Mode**.
```
Comment (rgerardi) [inline content/basics/vimmodes/insert.md]: > Just remember: if you need to move between words or lines, you need to switch back to **Normal Mode** by pressing the **ESC** key.

File: content/basics/vimmodes/insert.md [hunk 14]
```diff
@@ -4,4 +4,18 @@ date =  2017-10-30T10:48:58-04:00
...
+| **I**   | This key enter in insert mode begin line under the cursor						|
+| **a**   | This key enter in insert mode after under the cursor 								|
+| **A**   | This key enter in insert mode end line under the cursor  						|
+| **o**   | This key enter in insert mode on the next line under the cursor  		|
+| **O**   | This key enter in insert mode on the previous line under the cursor |
+
+------------
+
+> Just to remember, if you need to move between words or lines, you need to move to **Normal Mode**.  
+> To save file you need to type **ESC** *(Normal Mode)* then type **:w** and the most important to quit you type **:q**.
```
Comment (rgerardi) [inline content/basics/vimmodes/insert.md]: > To save the file, type the **ESC** key to switch back to **Normal Mode** then type **:w**. The most important: to quit **vim**, type **:q** while in **Normal Mode**.

File: content/basics/vimmodes/normal.md [hunk 15]
```diff
@@ -4,4 +4,10 @@ date =  2017-10-30T10:48:19-04:00
 weight = 5
 +++
 
-Normal Mode
+## Normal Mode  
+To enter in this mode, you need to *type* the **ESC** key.
```
Comment (rgerardi) [inline content/basics/vimmodes/normal.md]: To switch to this mode, *type* the **ESC** key.

File: content/basics/vimmodes/normal.md [hunk 16]
```diff
@@ -4,4 +4,10 @@ date =  2017-10-30T10:48:19-04:00
 weight = 5
 +++
 
-Normal Mode
+## Normal Mode  
+To enter in this mode, you need to *type* the **ESC** key.   
+This mode is where you will navigate in the file and you will can execute keys such dd(to remove the line on under the cursor, for example).
```
Comment (rgerardi) [inline content/basics/vimmodes/normal.md]: This mode is where you navigate around the file and you can execute commands such as **dd** to remove the line under the cursor, for example.

File: content/basics/vimmodes/normal.md [hunk 17]
```diff
@@ -4,4 +4,10 @@ date =  2017-10-30T10:48:19-04:00
...
+And you will do tasks such *copy,paste,cut,find/replace,etc...* we will see it in the next steps.
```
Comment (rgerardi) [inline content/basics/vimmodes/normal.md]: You can also perform tasks such as *copy, paste, cut, find/replace, etc...* We will see how in the following lessons.

File: content/basics/vimmodes/normal.md [hunk 18]
```diff
@@ -4,4 +4,10 @@ date =  2017-10-30T10:48:19-04:00
...
+### Command Mode
+To enter in this mode, you need to be in **Normal Mode** after that you need to *type* the **:** key.
```
Comment (rgerardi) [inline content/basics/vimmodes/normal.md]: To access this mode, you need to be in **Normal Mode**, then *type* the **:** key.

File: content/basics/vimmodes/normal.md [hunk 19]
```diff
@@ -4,4 +4,10 @@ date =  2017-10-30T10:48:19-04:00
...
 +++
 
-Normal Mode
+## Normal Mode  
+To enter in this mode, you need to *type* the **ESC** key.   
+This mode is where you will navigate in the file and you will can execute keys such dd(to remove the line on under the cursor, for example).
+And you will do tasks such *copy,paste,cut,find/replace,etc...* we will see it in the next steps.
+### Command Mode
+To enter in this mode, you need to be in **Normal Mode** after that you need to *type* the **:** key.  
+This mode is where you will execute commands such configurations on vim,save,quit,open new file,tabs,buffers,etc...
```
Comment (rgerardi) [inline content/basics/vimmodes/normal.md]: This mode is where you will execute commands such as updating **vim** configurations, saving, quitting, openning new files, tabs, buffers, etc...

File: content/basics/vimmodes/visual.md [hunk 20]
```diff
@@ -4,4 +4,14 @@ date =  2017-10-30T10:49:01-04:00
 weight = 7
 +++
 
-Visual Mode
+Visual Mode is where you can select words,lines,blocks,etc... There are some keys to enter in this mode, let's se them:
```
Comment (rgerardi) [inline content/basics/vimmodes/visual.md]: Visual Mode is where you can select words, lines, blocks, etc... While in **Normal Mode** you can use the following keys to switch to this mode:

File: content/basics/vimmodes/visual.md [hunk 21]
```diff
@@ -4,4 +4,14 @@ date =  2017-10-30T10:49:01-04:00
 weight = 7
 +++
 
-Visual Mode
+Visual Mode is where you can select words,lines,blocks,etc... There are some keys to enter in this mode, let's se them:  
+
+
+| Key          | Description   																											|
+|--------------|--------------------------------------------------------------------|
+| **v**        | This key enter will select the char under the cursor								|
```
Comment (rgerardi) [inline content/basics/vimmodes/visual.md]: This key switches to **Visual Mode** starting from the character under the cursor

File: content/basics/vimmodes/visual.md [hunk 22]
```diff
@@ -4,4 +4,14 @@ date =  2017-10-30T10:49:01-04:00
...
+| **V**   		 | This key enter will select the line under the cursor								|
```
Comment (rgerardi) [inline content/basics/vimmodes/visual.md]: This key switches to **Visual Mode** allowing selection of entire lines, starting with the current line

File: content/basics/vimmodes/visual.md [hunk 23]
```diff
@@ -4,4 +4,14 @@ date =  2017-10-30T10:49:01-04:00
...
+| **ctrl+v**   | This key enter will select in vertical block under the cursor 			|
```
Comment (rgerardi) [inline content/basics/vimmodes/visual.md]: This key switches to **Visual Mode** allowing selection of vertical blocks

File: content/basics/vimmodes/visual.md [hunk 24]
```diff
@@ -4,4 +4,14 @@ date =  2017-10-30T10:49:01-04:00
...
+
+------------
+> Just to remember, you can use all keys and combinations to select the text. We will see how navigate in the next step.
```
Comment (rgerardi) [inline content/basics/vimmodes/visual.md]: Just remember: you can use all navigation keys and combinations to select the text. We will see how to navigate in the next step.

**Review by rgerardi — COMMENTED** [review]
Looking good. A few suggestions to help with clarity and some minor spelling fixes. Please let me know what you think.

**Comment by renatosuero** [comment]
Thanks @rgerardi , I applied you suggestions =)

**Comment by rgerardi** [comment]
Excellent. Merging !

**Pull request merged**